### PR TITLE
Typify a bunch of Uniques.

### PR DESCRIPTION
**Pull request by will-ca**
Trying to mop up most of what's left. WIP-ish— Done but not much testing, will reorg, tweak spelling checker in light of, etc. No new deprecations/replacements/parametrizations, except for already dead but untypified Uniques/code. Submitting as draft to prevent duplicate work. May be easier to quickly review in current state though, as all the new UniqueTypes are in one block.

Will also get the overrides.

### CODE REVIEW

File: android/assets/jsons/Civ V - Gods & Kings/Beliefs.json [hunk 1]
```diff
@@ -275,7 +275,7 @@
     {
         "name": "Messiah",
         "type": "Enhancer",
-        "uniques": ["[+25]% Spread Religion Strength <for [Great Prophet] units>", "[-25]% Faith cost of generating Great Prophet equivalents", "[Faith] cost for [Great Prophet] units [-25]%"]
```
Comment (xlenstra) [inline android/assets/jsons/Civ V - Gods & Kings/Beliefs.json:278]: Why was this removed?

Comment (will-ca) [reply android/assets/jsons/Civ V - Gods & Kings/Beliefs.json:278]: As far as I was able to tell, its only appearance in the code was for AI automation, so it was already broken with no game rule effects, and I chose to deprecate and remove it for now rather than mixing a new implementation into this PR.

https://github.com/yairm210/Unciv/blob/6bc41a627c1118d8cb4736757b09515efe925ea6/core/src/com/unciv/logic/automation/ChooseBeliefsAutomation.kt#L180-L181

https://github.com/yairm210/Unciv/blob/3ce33fa1cfed1c01d6444a6bd1f39ce91979b7d3/core/src/com/unciv/models/ruleset/unique/UniqueType.kt#L847-L848

IMO Implementing it would best be a separate PR with more thorough testing and review than for the 1:1 conversion of existing logic I'm doing here.

Comment (xlenstra) [reply android/assets/jsons/Civ V - Gods & Kings/Beliefs.json:278]: If I'm not mistaken, it's already in the UniqueType as `BuyUnitsDiscount`

**Comment by yairm210** [comment]
Still relevant?

**Comment by JackRainy** [comment]
Looks similar to what I did in #6749

**Comment by xlenstra** [comment]
I'm voting to close this, as pretty much all uniques have been typed anyway, and getting the rest out of this PR isn't going to be easier than typing them again (I've never seen so many merge conflicts)

**Comment by SomeTroglodyte** [comment]
As will-ca did excellent groundwork but seems to be distracted at the moment, I cast may vote and declare it deciding. Kudos anyway and thanks for the inspiration.
